pokemongo_bot/cell_workers/sniper.py

```diff
@@ -325,86 +296,65 @@ def snipe(self, pokemon):
-                    nearby_stuff = self.bot.get_meta_cell()
-                    self.last_cell_check_time = time.time()
-
-                    # Retrieve nearby pokemons for validation
-                    nearby_pokemons.extend(nearby_stuff.get('wild_pokemons', []))
-                    nearby_pokemons.extend(nearby_stuff.get('catchable_pokemons', []))
-
-                    # Make sure the target really/still exists (nearby_pokemon key names are game-bound!)
-                    for nearby_pokemon in nearby_pokemons:
-                        nearby_pokemon_id = nearby_pokemon.get('pokemon_data', {}).get('pokemon_id') or nearby_pokemon.get('pokemon_id')
-
-                        # If we found the target, it exists and will very likely be encountered/caught (success)
-                        if nearby_pokemon_id == pokemon.get('pokemon_id', 0):
-                            exists = True
-                            success = True
-
-                            # Also, if the IDs arent valid, override them (nearby_pokemon key names are game-bound!) with game values
-                            if not pokemon.get('encounter_id') or not pokemon.get('spawn_point_id'):
-                                pokemon['encounter_id'] = nearby_pokemon['encounter_id']
-                                pokemon['spawn_point_id'] = nearby_pokemon['spawn_point_id']
-                            break
-
-                # If target exists, catch it, otherwise ignore
-                if exists:
-                    self._log('Yay! There really is a wild {} nearby!'.format(pokemon.get('pokemon_name')))
-                    self._teleport_back_and_catch(last_position, pokemon)
-                else:
-                    self._error('Damn! Its not here. Reasons: too far, caught, expired or fake data. Skipping...')
-                    self._teleport_back(last_position)
-                    
-                # Save target and unlock heartbeat calls
-                self._cache(uniqueid)
-                self.bot.hb_locked = False
+                self._error('Damn! Its not here. Reasons: too far, caught, expired or fake data. Skipping...')
+                self._teleport_back(last_position)
+
+            # Save target and unlock heartbeat calls
+            self._cache(pokemon)
+            self.bot.hb_locked = False
 
         return success
 
     def work(self):
-        # Do nothing if this task was invalidated
-        if self.disabled:
-            self._error("Sniper was disabled for some reason. Scroll up to find out.")
-            
-        elif self.bot.catch_disabled:
-            if not hasattr(self.bot,"sniper_disabled_global_warning") or \
-                        (hasattr(self.bot,"sniper_disabled_global_warning") and not self.bot.sniper_disabled_global_warning):
-                self._log("All catching tasks are currently disabled until {}. Sniper will resume when catching tasks are re-enabled".format(self.bot.catch_resume_at.strftime("%H:%M:%S")))
```

Comment:
Do not remove this! It was discussed and approved by multiple people and has proven EXTREMELY useful. If you do this, sniper will be the only catch task that does not respect the user's configuration.